kubernetes/templates/secret.yaml

```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ template "invidious.fullname" . }}
+  labels:
+    {{- include "invidious.labels" . | nindent 4 }}
+stringData:
+  INVIDIOUS_CONFIG: |
```

Comment:
Wouldn't it better to leave `INVIDIOUS_CONFIG` as a configmap, with secrets being templated in as needed? Having a separate `secrets.yaml` would also sensitive info to actually be encrypted, for example by using the [`helm-secrets`](https://github.com/jkroepke/helm-secrets) plugin, without making changes to `INVIDIOUS_CONFIG` needlessly difficult.

For example:
```yaml
# secrets.yaml
postgresql:
  auth:
    username: kemal
    password: kemal
  primary:
    initdb:
      username: kemal
      password: kemal

config:
  db:
    user: kemal
    password: kemal
  hmac_key: hmac_key
````


```yaml
# values.yaml
postgresql:
  enabled: true
  image:
    tag: 16
  auth:
    # username: SECRET
    # password: SECRET
    database: invidious
  primary:
    initdb:
      # username: SECRET
      # password: SECRET
      scriptsConfigMap: invidious-postgresql-init

config:
  db:
    # user: kemal
    # password: kemal
    host: invidious-postgresql
    port: 5432
    dbname: invidious
  # hmac_key: SECRET
```

Reply:
I do agree on that.

Reply:
Well, I wanted to do the same but I lack the idea how to do this properly. It would be easier if config options would be a separate env vars, but this is one env var and we are unable to concatenate it in helm chart before pushing it to the pod/container.

Reply:
Also the solution here allows the usage of helm-secret without any difficulty - it's an object, not a string in the values, so you can split the secret parts to other values file and encrypt it with helm-secrets using SOPS or provide it externally using vals.

Reply:
`Secret`s aren't encrypted by default, but `ConfigMap`s cannot be encrypted at all.
If we have an idea how we can split part of config to go to `ConfigMap` and `Secret` we can then roll it back to `ConfigMap` and selectively move parts of config values to `Secret`. If we have no idea how to do this, I'd personally prefer to have whole `INVIDIOUS_CONFIG` env var in the `Secret` due to sensitive values inside.